store/storetest/user_store.go

```diff
@@ -6010,3 +6011,160 @@ func testGetUsersWithInvalidEmails(t *testing.T, ss store.Store) {
 	require.NoError(t, err)
 	assert.Len(t, users, 1)
 }
+
+func testUserStoreSearchUsersMultilingual(t *testing.T, ss store.Store, s SqlStore) {
+	u1 := &model.User{
+		Username:  "test1" + model.NewId(),
+		FirstName: "Inígo",
+		LastName:  "Martínez",
+		Nickname:  "Berridi",
+		Email:     MakeEmail(),
+	}
+	_, err := ss.User().Save(u1)
+	require.NoError(t, err)
+	defer func() { require.NoError(t, ss.User().PermanentDelete(u1.Id)) }()
+
+	u2 := &model.User{
+		Username:  "test2" + model.NewId(),
+		FirstName: "Zinëdìne",
+		LastName:  "Zidanë",
+		Email:     MakeEmail(),
+	}
+	_, err = ss.User().Save(u2)
+	require.NoError(t, err)
+	defer func() { require.NoError(t, ss.User().PermanentDelete(u2.Id)) }()
+
+	u3 := &model.User{
+		Username:  "test3" + model.NewId(),
+		FirstName: "Thomas ",
+		LastName:  "Müller",
+		Nickname:  "Fußballspieler",
+		Email:     MakeEmail(),
+	}
+	_, err = ss.User().Save(u3)
+	require.NoError(t, err)
+	defer func() { require.NoError(t, ss.User().PermanentDelete(u3.Id)) }()
+
+	u4 := &model.User{
+		Username:  "test4" + model.NewId(),
+		FirstName: "Jérémie ",
+		LastName:  "Jéry",
+		Email:     MakeEmail(),
+	}
+	_, err = ss.User().Save(u4)
+	require.NoError(t, err)
+	defer func() { require.NoError(t, ss.User().PermanentDelete(u4.Id)) }()
+
+	// The users returned from the database will have AuthData as an empty string.
+	nilAuthData := new(string)
+	*nilAuthData = ""
+	u1.AuthData = nilAuthData
+	u2.AuthData = nilAuthData
+	u3.AuthData = nilAuthData
+	u4.AuthData = nilAuthData
```

Comment:
What is the reason for this?

Reply:
It seems to be a necessary action in order for the test to correctly identify the user. That is because the `u` (user) objects that the tests are being validated against have `nil` as `AuthData` but the database objects have `AuthData` equal to empty string. So with this way the validating objects' `AuthData` are being initialized as empty strings too